### PR TITLE
buttons can be passed icons, style variants, dynamic component loading

### PR DESCRIPTION
**Pull request by stuartromanek**

### CODE REVIEW

File: src/components/button/AposButton.vue [hunk 1]
```diff
@@ -15,24 +20,43 @@ export default {
...
+          modifiers += ' apos-button--icon-only';
+        } else {
+          modifiers = 'apos-button--icon-only';
+        }
+      }
+      return modifiers;
+    },
+    iconComponent () {
+      if (this.icon) {
+        return () => import(`vue-material-design-icons/${this.icon}.vue`);
```
Comment (abea) [inline src/components/button/AposButton.vue]: :whoah:

File: src/components/button/AposButton.vue [hunk 2]
```diff
@@ -91,6 +114,11 @@ export default {
     }
   }
 
+  .apos-button--hollow,
```
Comment (abea) [inline src/components/button/AposButton.vue]: "hollow" doesn't make total sense to me, but not a biggie.

File: src/components/button/AposButton.vue [hunk 3]
```diff
@@ -166,17 +194,36 @@ export default {
...
-    .apos-button__label {
+    .apos-button__content {
       opacity: 0;
     }
     .apos-loading {
       opacity: 1;
     }
   }
 
+  .apos-button--icon-only {
```
Comment (abea) [inline src/components/button/AposButton.vue]: Could make this terser as `.apos-button--icon`.

File: src/components/button/AposButton.vue (hunk 2 above)
Comment (abea) [reply src/components/button/AposButton.vue]: "clear"?

Comment (stuartromanek) [reply src/components/button/AposButton.vue]: hmm i thought hollow and ghost we're pretty common .. looks like Zurb calls them hollow, bootstrap calls them outline. I think i like outline of these

Comment (abea) [reply src/components/button/AposButton.vue]: Ah, I never really spent any time with Foundation.

**Pull request merged**